gtfu/src/main/java/gtfu/GraphicReport.java

```diff
@@ -274,7 +275,11 @@ private void addRandomTestData(float prob) {
                     }
 
                     gpsMap.put(id,latLonMap);
-                    TripReportData td = new TripReportData(id, t.getHeadsign(), start, duration, "test", "test", "test", new Stats(gpsMap.get(id).values()));
+                    // Create lists as input to Stats
+                    List<Double> updateFreqList = gpsMap.get(id).values().stream().map(GPSData::getSecsSinceLastUpdateDouble).filter(secs -> secs > 0).collect(Collectors.toList());
```

Comment:
I'm not a fan of these constructs: they're long, convoluted and when you get an exception, it's difficult to see which part actually failed. But you built the whole design of this PR around them, so to not have them you would have to change all kinds of things.

Reply:
I wrote these originally as loops, and would be happy to revert. I'm curious for your take on which is preferable:

Previous code:
```            List<Double> updateFreqList = new ArrayList<>();
            List<Double> updateTimeList = new ArrayList<>();

            for (GPSData gpsData : gpsMap.get(id).values()) {

                if(gpsData.secsSinceLastUpdate > 0) {
                    Double secsSinceLastUpdateDouble = Double.valueOf(gpsData.secsSinceLastUpdate);
                    updateFreqList.add(secsSinceLastUpdateDouble);
                }

                updateTimeList.add(Double.valueOf(gpsData.millis));
            }```